File: tests/sigterm-delay/sample_app.go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"syscall"
	"time"
)

func main() {
	sigs := make(chan os.Signal, 1)
	signal.Notify(sigs, syscall.SIGTERM)
	fmt.Println("Go app: started")
	select {
	case <-time.After(5 * time.Second):
		fmt.Println("Go app: delayed message")
	case <-sigs:
		fmt.Println("Go app: captured SIGTERM")
		os.Exit(143)
	}
	<-sigs
	fmt.Println("Go app: captured SIGTERM")
	os.Exit(143)
}

File: tests/sigterm-delay/SampleApp.java
public class SampleApp {
    public static void main(String[] args) throws Exception {
        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
            System.out.println("Java app: captured SIGTERM");
        }));
        System.out.println("Java app: started");
        Thread.sleep(5000);
        System.out.println("Java app: delayed message");
        while (true) {
            Thread.sleep(1000);
        }
    }
}
